src/OptionSettlementEngine.sol

```diff
@@ -530,11 +545,14 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
     //////////////////////////////////////////////////////////////*/
 
     /// @dev Internal helper function to calculate, record, and emit event for fee accrual
-    /// when writing (on underlying asset) and when exercising (on exercise asset)
+    /// when writing (on underlying asset) and when exercising (on exercise asset). Checks
+    /// that fee switch is enabled, otherwise returns fee of 0 and does not record or emit.
     function _calculateRecordAndEmitFee(address assetAddress, uint256 assetAmount) internal returns (uint256 fee) {
-        fee = ((assetAmount * feeBps) / 10_000);
-        feeBalance[assetAddress] += fee;
-        emit FeeAccrued(assetAddress, msg.sender, fee);
+        if (isFeeSwitchEnabled) {
```

Comment:
I put the check inside the internal function, but it might be cleaner to move to the call sites (write and exercise).

Reply:
I think keeping it here is probably more maintainable

Reply:
Sounds good. Leaving where is